hack/dockerfiles/lint.Dockerfile

```diff
@@ -18,8 +18,9 @@ WORKDIR /go/src/github.com/moby/buildkit
 FROM base as golangci-lint
 ARG BUILDTAGS
 RUN --mount=target=/go/src/github.com/moby/buildkit --mount=target=/root/.cache,type=cache,sharing=locked \
-  GOARCH=amd64 golangci-lint run --build-tags "${BUILDTAGS}" && \
-  GOARCH=arm64 golangci-lint run --build-tags "${BUILDTAGS}" && \
+  GOOS=linux GOARCH=amd64 golangci-lint run --build-tags "${BUILDTAGS}" && \
+  GOOS=windows GOARCH=amd64 golangci-lint run --build-tags "${BUILDTAGS}" && \
+  GOOS=linux GOARCH=arm64 golangci-lint run --build-tags "${BUILDTAGS}" && \
```

Comment:
Should we do freebsd/amd64 too?

Reply:
Yeahhh, why not! :tada: Done in eb32ebf

Reply:
Hmm. The lint run took 20 minutes. Is that just because it's establishing a cache for the new architectures, or would it be worthwhile to split the lint into multiple jobs?

Reply:
Yeah, we've essentially doubled the amount of linting we do now - it already takes around 10min on master (also *what* why is it so slow?).

Splitting SGTM, I'm a bit pressed for time atm, so might not get to it super soon.